packages/batch-submitter/src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -252,25 +259,30 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
     })
 
     const nonce = await this.signer.getTransactionCount()
+    // Generate the transaction we will repeatedly submit
+    const tx = await this.chainContract.customPopulateTransaction.appendSequencerBatch(
+      batchParams
+    )
     const contractFunction = async (gasPrice): Promise<TransactionReceipt> => {
       this.logger.info('Submitting appendSequencerBatch transaction', {
         gasPrice,
         nonce,
         contractAddr: this.chainContract.address,
       })
-      const tx = await this.chainContract.appendSequencerBatch(batchParams, {
+      const txResponse = await this.signer.sendTransaction({
+        ...tx,
         nonce,
         gasPrice,
       })
       this.logger.info('Submitted appendSequencerBatch transaction', {
-        txHash: tx.hash,
-        from: tx.from,
+        txHash: txResponse.hash,
+        from: txResponse.from,
```

Comment:
It may be nice to log `gasUsed` here and also `status` to learn if it reverts

Reply:
`gasUsed` and `status` are only available on the tx receipt. We log the full tx receipt here: https://github.com/ethereum-optimism/optimism/blob/cc742715ecbca98248367d67f51a5f03038f5ba2/packages/batch-submitter/src/batch-submitter/batch-submitter.ts#L241